src/data-handler/binned-data-fetcher.js

```diff
@@ -87,8 +87,13 @@ export default function createBinnedDataFetcher({ layoutService, extremumModel,
       return maxBinDensity;
     },
     fetch: () => {
-      if (!layoutService.meta.isBigData || layoutService.meta.isSnapshot) {
-        return Promise.resolve(layoutService.getLayoutValue('dataPages'));
+      if (layoutService.meta.isSnapshot) {
+        const dataPages = layoutService.getLayoutValue('dataPages');
+        if (dataPages[1]) {
+          populateBins(dataPages);
```

Comment:
While you are doing changes in the code here, could you change the following in `populateBins`:
* Rename the input parameter `dp` to `dataPages`
* Remove the comment that says "Hack for snapshot"

Reply:
Renamed and removed.